### DISCUSSION TITLE
RedHat/Fedora port

### DISCUSSION THREAD

**Discussion by johnkerl** [johnkerl/miller#1329] [Announcements]
Miller is in a lot of Linux/BSD/Homebrew distros but an explicit absence is Centos/RedHat/Fedora. There's more paperwork associated with that than I've had the gumption to deal with. If anyone reading this is an expert in that area ... I'd love to chat. I have an SRPM ready to go. :)

**Comment by skitt** [comment]
I can help with this, at least on the Fedora side. Are you looking for help getting your package in (so you’d be the Fedora maintainer, possibly with a co-maintainer), or for someone to take care of the package in Fedora entirely (so based on your SRPM, but you wouldn’t be the Fedora maintainer)?

**Comment by johnkerl** [comment]
Either-or, really --  whatever's easiest for everyone. On all the other arches I'm not the primary maintainer within the distro.

**Comment by skitt** [comment]
OK, I’ll take care of it (in the coming weeks).

**Comment by johnkerl** [comment]
@skitt thank you!

**Comment by bandikoot** [comment]
Hello, any updates here? I'm interested in miller package for Fedora, too (:

**Comment by skitt** [comment]
Oops, yes, sorry, I got rather side-tracked (for a year...), I’ll get onto it.

**Comment by skitt** [comment]
Far too much time later, https://koji.fedoraproject.org/koji/taskinfo?taskID=34109609 — so I’ll be able to file a request for review to get this into Fedora.

**Comment by johnkerl** [comment]
thanks @skitt!!!

**Comment by afranke** [comment]
Will this hit F30 as well @skitt ? The build you link to is for rawhide/fc31.

**Comment by skitt** [comment]
> Will this hit F30 as well @skitt ? The build you link to is for rawhide/fc31.

Once I get it sponsored I can push it to other branches (30, 29...).

**Comment by skitt** [comment]
John, your `.spec` file currently uses the `mlr` package name, but I use `miller` in Debian (which matches the repo name). Would you prefer the Fedora package to be called `mlr` too, or can I use the `miller` name?

**Comment by johnkerl** [comment]
@skitt yeah the tool so nice I named it twice. :^/ Every once in a while I wish I'd called the package and the executable both `miller`, sigh. Anyway.

As long as the executable is still called `mlr` then the docs won't confuse people, is the crucial thing.

If it's just the name of the package being `miller` containing an executable called `mlr`, then let's keep doing what you're already doing for Debian -- as the path of least surprise for everyone.

Thanks!!!

**Comment by skitt** [comment]
The `miller` package has been reviewed and approved. The remaining steps for it to get into Fedora are the creation of a repository for the packaging (https://pagure.io/releng/fedora-scm-requests/issue/12323), then me uploading it, building and finally waiting for it to be old enough to get into the archives.

**Comment by skitt** [comment]
`miller` is now in Fedora, https://koji.fedoraproject.org/koji/packageinfo?packageID=28934 — currently FC31, I’ll add it to FC30 and FC29. @johnkerl, I think this issue can be closed, at long last ;-).

**Comment by johnkerl** [comment]
thank you @skitt !!!!!!!!! :)

**Comment by johnkerl** [comment]
i'll leave this open until i include a link on the main miller github page, which points to other distros.

**Comment by johnkerl** [comment]
done -- thanks again @skitt !! :)

**Comment by johnkerl** [comment]
@skitt -- I have a couple open dependabot PRs which fail build b/c I currently have go 1.15 in `go.mod`. (These aren't crucial -- I can cancel the PRs, no problem.)

Meanwhile I seem to recall a while back I tried a newer Go compiler version in `go.mod` but it caused Centos/RH7 issues since not all platforms use the latest (or even near-latest) Go compiler.

Do you perchance have any guidance on what would be a good minimal Go compiler version to depend on today -- is it still 1.15?

**Comment by skitt** [comment]
On RHEL 7 (and CentOS), for decent Go support you really need to use the Go toolset from Software Collections, and 1.18 is available there. RHEL 8 has Go 1.18 by default. That seems like a decent minimum version (although Go 1.18 reached EOL a month ago!).

**Comment by johnkerl** [comment]
Thanks @skitt ! I'll research how to get the Go toolset from Software Collections integrated into the Miller repo.

**Comment by skitt** [comment]
See https://developers.redhat.com/HW/Go-RHEL-7 for details.

**Comment by johnkerl** [comment]
@skitt we're now building with Go 1.18.

Is there anything remaining to do here?

**Comment by skitt** [comment]
I haven’t finished getting Go-based Miller into Fedora again, I need to package `facette/natsort` and `lumin`.

**Comment by johnkerl** [comment]
Thanks @skitt !
